galore/cross_sections.py

```diff
@@ -281,3 +303,242 @@ def _eval_fit(energy, coeffs):
+        
+        Note: subshells are like ('2p1/2', '2p3/2', '3p1/2', '3p3/2',...) for obital p
+    
+    Returns:
+        avg_orbital_cross_sections(np.array): average cross sections of obitals 
+        
+        Note: the above array are like ('2p', '3p',...) for obital p
+    """
+    
+    
+    subshells_cross_section_arrays =  np.array_split(subshells_cross_sections, 2)
+    electrons_numbers_arrays = np.array_split(electrons_number, 2)
+    
+    subshells_cross_sections_sum = np.array(list(map(sum,subshells_cross_section_arrays)))
+    
+    subshells_electrons_number_sum = np.array(list(map(sum,electrons_numbers_arrays)))
+    
+    avg_orbital_cross_sections = np.true_divide(subshells_cross_sections_sum,subshells_electrons_number_sum)
+    
+        
+    return avg_orbital_cross_sections
+
+def _cross_sections_from_csv_data(energy,data,dataset):
+    """
+    Get cross sections from data dict 
+    For known sources, data is based on tabulation of Yeh/Lindau (1985).[1]
+    Otherwise, energies in keV from 1-1500 are used with log-log polynomial
+    parametrisation of data from Scofield.[2]
+    
+    References:
+      1. Yeh, J.J. and Lindau, I. (1985)
+         Atomic Data and Nuclear Data Tables 32 pp 1-155
+      2. J. H. Scofield (1973) Lawrence Livermore National Laboratory
+         Report No. UCRL-51326
+    
+    Args:
+        energy(float): energy value  
+        data(dict): data from read_csv_file()
+        reference(str): 'Scofield' or 'Yeh'
+        
+    Returns:
+        orbitals_cross_sections_dict: containing orbitals 's', 'p', 'd', 'f' and 
+                                      maximum cross sections of each orbital. 
+                                      Missing data is represented as None.
+  
+    """   
+
+    n_subshells = len(data['electron_counts'])
+    subshells_headers = data['headers'][-n_subshells:]
+    cross_sections_by_subshells = data['data_table'].T[-n_subshells:]
+    
+    ##build dicts for easy calculation. 
+    electron_counts_by_subshells = dict(zip(subshells_headers,data['electron_counts']))
+    cross_sections_by_subshells = dict(zip(subshells_headers,data['data_table'].T[-n_subshells:]))
+
+    ## match the import energy
+    energy_index = np.abs(data['data_table'].T[0] - float(energy)).argmin()
+
+    ## build result dict
+    orbitals_cross_sections_dict = {}
+    
+    ## result for s orbital
+    s_cross_sections = np.array([value[energy_index] for key, value in cross_sections_by_subshells.items() if 's' in key])
+    electrons_number = np.array([value for key, value in electron_counts_by_subshells.items() if 's' in key])
+    ## get unit cross sections of s orbital
+    unit_cross_sections = s_cross_sections/electrons_number
+    ## get highest cross section of obital s
+    highest_obital_cross_section = np.max(np.nan_to_num(unit_cross_sections))
+    orbitals_cross_sections_dict['s'] = highest_obital_cross_section
```

Comment:
This isn't necessarily the correct data to use. We want the "highest" orbital from 1s, 2s, 3s... which is not necessarily the one with the largest scattering cross section.